### PR TITLE
src/service.c: Create a fake primary device icon on Lomiri if no device is present

### PR DESCRIPTION
**Pull request by tari01**
fixes https://gitlab.com/ubports/development/core/packaging/ayatana-indicators/ayatana-indicator-power/-/issues/2

### CODE REVIEW

**Comment by gber** [comment]
This works as in it shows a full battery. I'm wondering if it would be better to show an icon for AC power connection in this case?

**Comment by tari01** [comment]
Normally, the indicator is not shown at all if there is no primary device. Since it does show in the PDK, I figured it would be best to show people what to expect on a physical handheld. Of course, we can do whatever else we want: show an AC adapter icon, use some kind of disabled icon, or simply not show anything.

Mind you, we already use this approach (i.e. show what it would be like on a phone) for a bunch of other stuff that do nothing: Bluetooth, Rotation, etc.

**Comment by sunweaver** [comment]
I'd say let's show an AC-like icon or a battery icon strike through (whatever we have that resembles a fake battery).

**Comment by tari01** [comment]
Here's my last idea: a non-device "power" icon (something like a lightning bolt). Then we would have no need to disable it or strike it through.

I think we have enough to choose from. @gber , @JamiKettunen, @sunweaver: please cast your vote :)

**Comment by sunweaver** [comment]
How about battery-missing-symbolic.svg?

**Comment by gber** [comment]
> How about battery-missing-symbolic.svg?

Sounds sensible for the PDK case. So in "regular" desktop mode the indicator will now be visible at all, right?

**Comment by tari01** [comment]
> How about battery-missing-symbolic.svg?

I vote for this.

**Comment by tari01** [comment]
> So in "regular" desktop mode the indicator will now be visible at all, right?

On Lomiri, the visibility is governed by 3 GSettings values (device present / device charging / never).

**Comment by JamiKettunen** [comment]
If no power icon is shown on a normal desktop Lomiri session (not PDK) when there's no battery device to report details about I don't really mind what the selected PDK behavior is especially since the other indicators already seem to be shown anyway :)

**Comment by tari01** [comment]
@sunweaver: Amended. Mergification, please.

File: src/service.c [hunk 1]
```diff
@@ -494,6 +494,16 @@ create_header_state (IndicatorPowerService * self)
           g_object_unref (icon);
         }
     }
+    else if (ayatana_common_utils_is_lomiri())
```
Comment (sunweaver) [inline src/service.c:497]: So, this always kicks in if Lomiri runs on a device without reported batteries. As we are mainlining Lomir + Indicators to be suitable for desktop and touch based systems for any distro, so also with the use case of running on desktops without batteries attached, we should probably find a way to check if we are in the PDK environment or similar. Right?

**Review by sunweaver — COMMENTED** [review]
Question about use cases, see inline comment.

File: src/service.c (hunk 1 above)
Comment (tari01) [reply src/service.c:497]: > So, this always kicks in if Lomiri runs on a device without reported batteries. As we are mainlining Lomir + Indicators to be suitable for desktop and touch based systems for any distro, so also with the use case of running on desktops without batteries attached, we should probably find a way to check if we are in the PDK environment or similar. Right?

That should not be necessary. The newly added case does not control the visibility of the indicator. We are only exporting all the header properties to D-Bus. The visibility is controled by GSettings here:

https://github.com/AyatanaIndicators/ayatana-indicator-power/blob/8a223541810566c2c5ccc5d4f7f9f0b2425bf0ca/src/service.c#L452-L453

and here:

https://github.com/AyatanaIndicators/ayatana-indicator-power/blob/8a223541810566c2c5ccc5d4f7f9f0b2425bf0ca/src/service.c#L402-L438

Under normal circumstances **no batteries == no indicator**.

Comment (sunweaver) [reply src/service.c:497]: I currently have a VM here, that has no battery. And Lomiri shows the indicator nonetheless (as cogwheel).
image

**Comment by sunweaver** [comment]
I have now merged this, as battery-missing in lomiri without battery installed is better than the cogwheel. However, this needs love in Lomiri I sense.

**Comment by tari01** [comment]
> I have now merged this, as battery-missing in lomiri without battery installed is better than the cogwheel. However, this needs love in Lomiri I sense.

I have a feeling that Lomiri is not using the Gsettings that set when to show the indicator.

**Pull request merged**